### ISSUE TITLE
On changing planes while carrying the flag

### ISSUE THREAD

**Issue by AirmashPewPewPew** [wight-airmash/ab-server#15]
I broke something.

I joined the EU CTF2 server through airmash.online, positioned on blue team. Bots are unbalanced; seems like there's only three bots on red. Blue is killing it: 2-0. I seem to be the only human, which means it's time to fuck it all up. I'm going with my trusty Mohawk and taking over as leader...

Immediately, I notice Yutu-2 is carrying the flag back to blue base, the red flag fluttering over the Atlantic, as if mocking the red, Communist regime. She's running out of fuel, so I to assist. "#drop," I say, and she relinquishes the flag atop my copter propellers. The rest of the squad approaches. "On your orders!" says Mars 3. However, I feel the weight of the flag on my small copter frame, and I decide to stand still for two seconds. I change planes to Goliath...

... and it works! I appear at the blue re-spawn point, transformed into the largest plane in the fleet, the red flag still on my back. I'm not as fast as I thought I would be, but no matter. It's time to mock the enemy -- taunt them with our inevitable victory -- by destroying them mercilessly while I carry their flag. The blue air force and I approach the red base to vanquish the three enemies.

Red plane after red plane spawns; we're in their own base, keeping them at bay. Suddenly, Yutu -- Yutu 2's evil twin -- shoots a missile in my back and takes me down. But luck was on my side. When the smoke settled, and I re-spawned at my base, I still had the enemy flag on my back!

I'm running out of story to describes the rest of the bug report, so

TL;DR

1. If a player changes planes while she's carrying the flag, she will continue to carry the flag
2. If a player dies after she performs 1, she will continue to carry the flag. She may die multiple times; it doesn't seem to matter.
3. If, after 1, a player transitions into spectator mode, she will continue to carry the flag. The flag appears in the top-left corner of the map. After re-joining the game from spectator mode, she will continue to carry the flag.
4. If, after 1, a player disconnects and immediately re-connects, and if the server hands the points back to the player, the player will continue to carry the flag.
5. At some point, the player may re-join, but the server does not give her points back. The enemy flag will be locked to the ground, and no one will be able to pick it up.

The server seems to be in state 5.

<img width="1426" alt="Screen Shot 2019-08-30 at 10 05 13 PM" src="https://user-images.githubusercontent.com/46804081/64058172-3c9c5000-cb75-11e9-9bd6-f5b2dede831e.png">

My bad,

pew pew pew

**Comment by janwillemb** [comment]
:) nice
